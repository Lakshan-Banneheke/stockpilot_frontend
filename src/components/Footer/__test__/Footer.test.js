import { render, screen } from '@testing-library/react'
import Footer from '../Footer'
import '@testing-library/jest-dom/extend-expect'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

const MockFooter = () => {
  return (
    <BrowserRouter>
      <Footer />
    </BrowserRouter>
  )
}

describe('Footer', () => {
  it('renders the slogan', async () => {
    render(<MockFooter />)
    const divElement = screen.getByText(
      /The website for all Your Crypto and Stock Analysis needs/i
    )
    expect(divElement).toBeInTheDocument()
  })

  it('renders links', async () => {
    render(<MockFooter />)
    const divElement = await screen.findByText(/Helpful Links/i)
    expect(divElement).toBeInTheDocument()
  })

  it('renders footer without crashing', async () => {
    render(<MockFooter />)
    const divElement = screen.getByTestId('header')
    expect(divElement).toBeTruthy()
  })

  it('renders and passed snapshot', async () => {
    const tree = renderer.create(<MockFooter />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
