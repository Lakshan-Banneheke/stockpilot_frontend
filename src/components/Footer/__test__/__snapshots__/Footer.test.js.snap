// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders and passed snapshot 1`] = `
<div
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <div
    className="MuiBox-root MuiBox-root-32"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <a
          className="MuiGrid-root"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            className="makeStyles-logo-27"
            src="logo.svg"
          />
        </a>
        <p
          className="MuiTypography-root MuiTypography-body1"
          data-testid="header"
        >
          The website for all Your Crypto and Stock Analysis needs
        </p>
      </div>
      <div
        className="MuiGrid-root makeStyles-gridItem-25 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Helpful Links
        </p>
        <div
          className="makeStyles-gridItemLinks-26"
        >
          <a
            className="makeStyles-link-28"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="makeStyles-link-28"
            href="/analyze/crypto"
            onClick={[Function]}
          >
            Crypto
          </a>
          <a
            className="makeStyles-link-28"
            href="/analyze/stock"
            onClick={[Function]}
          >
            Stock
          </a>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-gridItem-25 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Contact Us
        </p>
        <div
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-29"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-29"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-29"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
            />
          </svg>
        </div>
        <div
          className="MuiGrid-root makeStyles-mail-31 MuiGrid-container"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "marginRight": 20,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            32/1, Temple Road, Colombo 7
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container"
          style={
            Object {
              "marginTop": 15,
            }
          }
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "marginRight": 20,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            info.stockpilot@gmail.com
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
